lib/net.js

```diff
@@ -319,14 +319,25 @@ Socket.prototype._onTimeout = function() {
 
 
 Socket.prototype.setNoDelay = function(enable) {
+  if (!this._handle) {
+    this.once('connect',
+              enable ? this.setNoDelay : this.setNoDelay.bind(this, enable));
```

Comment:
Is there a reason why the ternary is here instead of just passing in `enable`? Is it the `.bind` perf?